code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -31,85 +68,207 @@
 	ranged = 1
 	smart = TRUE
 
 /mob/living/simple_animal/hostile/hivebot/range/rapid
+	projectiletype = /obj/item/projectile/bullet/pistol/hivebotspike/needle
 	rapid = 1
 
-/mob/living/simple_animal/hostile/hivebot/range/strong
-	name = "Strong Hivebot"
-	desc = "A robot, this one is armed and looks tough!"
-	health = 80
+//Creates a reference to its parent beacon on init.
+/mob/living/simple_animal/hostile/hivebot/Initialize(mapload,mob/living/simple_animal/hostile/hivebot/hivebotbeacon)
+	if(hivebotbeacon)
+		linked_parent = hivebotbeacon
+		linked_parent.existing_bots += 1
+	.=..()
 
 /mob/living/simple_animal/hostile/hivebot/death()
-	..()
-	visible_message("<b>[src]</b> blows apart!")
-	new /obj/effect/decal/cleanable/blood/gibs/robot(src.loc)
-	spark(src, 3, alldirs)
+	..(null,"blows apart!")
+	if(linked_parent)
+		linked_parent.existing_bots -= 1
+		linked_parent.linked_bots -= src
+	var/T = get_turf(src)
+	new /obj/effect/gibspawner/robot(T)
+	spark(T, 1, alldirs)
 	qdel(src)
 	return
 
-/mob/living/simple_animal/hostile/hivebot/tele//this still needs work
-	name = "Beacon"
-	desc = "Some odd beacon thing"
-	icon_state = "def_radar-off"
-	icon_living = "def_radar-off"
+/mob/living/simple_animal/hostile/hivebot/think()
+	. =..()
+	if(stance == HOSTILE_STANCE_IDLE)
+		icon_state = "[initial(icon_state)]"
+	else
+		icon_state = "[initial(icon_state)]_armed"
+
+/mob/living/simple_animal/hostile/hivebot/proc/wakeup()
+	stance = HOSTILE_STANCE_IDLE
+
+/mob/living/simple_animal/hostile/hivebot/emp_act(severity)
+	LoseTarget()
+	stance = HOSTILE_STANCE_TIRED
+	addtimer(CALLBACK(src, .proc/wakeup), 150)
+	if(severity == 1.0)
+		apply_damage(10)
+
+/mob/living/simple_animal/hostile/hivebotbeacon
+	name = "Hivebot beacon"
+	desc = "An odd and primitive looking machine. It emanates of strange and powerful energies. It bears no manufacturer markings of any kind."
+	icon = 'icons/mob/npc/hivebot.dmi'
+	icon_state = "hivebotbeacon_active"
+	icon_living = "hivebotbeacon_active"
 	health = 200
 	maxHealth = 200
+	projectilesound = 'sound/weapons/taser2.ogg'
+	projectiletype = /obj/item/projectile/beam/hivebotdischarge
+	wander = 0
 	status_flags = 0
-	anchored = 1
-	stop_automated_movement = 1
-	var/bot_type = NORMAL // type of bot, 1 is normal, 2 is ranged, 3 is rapid rnaged.
-	var/bot_amt = 10
-	var/spawn_delay = 600
+	faction = "hivebot"
+	ranged = 1
+	rapid = 1
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	minbodytemp = 0
+	speed = -10
+	see_in_dark = 8
+	smart = 1
+	destroy_surroundings = 0
+	var/bot_type = NORMAL // type of bot, 1 is normal, 2 is ranged, 3 is rapid ranged
+	var/bot_amt = 48 //Number of total bots that are spawned before the beacon disappears completely.
+	var/max_bots = 16 //Number of bots linked to this beacon specifically that can exist, before spawning more is halted.
+	var/list/linked_bots = list()
+	var/latest_child = null
+	var/existing_bots = 0

```

Comment:
```suggestion
```

This variable is unnecessary and can be substituted with `linked_bots.len()`, since that retreives the amount of bots linked.